daemon/oci_linux_test.go

```diff
@@ -48,6 +51,18 @@ func setupFakeDaemon(t *testing.T, c *container.Container) *Daemon {
 		c.NetworkSettings = &network.Settings{Networks: make(map[string]*network.EndpointSettings)}
 	}
 
+	// HORRIBLE HACK: clean up shm mounts leaked by some tests. Otherwise the
+	// offending tests would fail due to the mounts blocking the temporary
+	// directory from being cleaned up.
```

Comment:
I checked: only the tests are impacted. The shm mounts are not leaked on container exit—though I'm not sure where in the code they get unmounted.

Reply:
I think I was running into this one as well in https://github.com/moby/moby/pull/45474#discussion_r1197048332
and https://github.com/moby/moby/pull/43346/files#diff-7044cd74671fe1aa629c3374b3b65fb57e7d47c6a9a20a23b539ee1425a88f2eL19-L52


(Not sure why I didn't use `c.ShmPath` though 🤔